src/app/models/field/emailField.ts

```diff
@@ -39,6 +39,41 @@ const createEmailFieldSchema = () => {
       type: Boolean,
       default: false,
     },
+    hasAllowedEmailDomains: {
+      type: Boolean,
+      default: false,
+    },
+    allowedEmailDomains: {
+      type: [String],
```

Comment:
Might want to add the `trim` option to prevent strings with whitespace, and the `match` option here instead of inside the validator:

```ts
allowedEmailDomains: {
  type: [
    {
      type: String,
      trim: true,
      match: [/.+\..+/, 'Please fill a valid email domain.'],
    },
  ],
  ...
},
```

Similar to how emailDomain is validated in `agency.server.model`